### PR TITLE
Bgp evpn

### PR DESCRIPTION
**Pull request by amrghosh**
EVPN VXLAN Data modelling updted REview - 2
[LINK](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/bgp_evpn/artifacts/openapi.yaml)

### CODE REVIEW

File: device/bgp/evpn/broadcastdomainv4.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  schemas:
+    Bgp.BroadcastDomainVxlanV4:
+      x-include:      
+      - '../../device.yaml#/components/schemas/Device.Active'
+      - './broadcastdomain.yaml#/components/schemas/Bgp.BroadcastDomain'
+      description: >-
+        List of BroadcastDomains per EVI
+      type: object
+      properties: 
+        mac_ranges:
```
Comment (manomugdha) [inline device/bgp/evpn/broadcastdomainv4.yaml:11]: should not this be part of base yaml?

File: device/bgp/evpn/broadcastdomainv6.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+components:
+  schemas:
+    Bgp.BroadcastDomainVxlanV6:
+      x-include:      
+      - '../../device.yaml#/components/schemas/Device.Active'
+      - './broadcastdomain.yaml#/components/schemas/Bgp.BroadcastDomain'
```
Comment (manomugdha) [inline device/bgp/evpn/broadcastdomainv6.yaml:6]: same comments as v4.

File: device/bgp/evpn/ethernetsegment.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      properties:        
+        esi_type:
+          description: >-
+            RFC 7432 defines several types of identifiers, 
+            with explicit rules for the ESI value assignment. 
+            Use this field to select the desired ESI type .
+          type: string
+        esi:
+          description: >-
+            10-octet Ethernet segment identifier (esi)
```
Comment (manomugdha) [inline device/bgp/evpn/ethernetsegment.yaml:16]: it is 10 or 9 octets?

File: device/bgp/evpn/ethernetsegment.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          minimum: 0
+          maximum: 16777215
+          default: 0
+        advertise_ad_evi_route:
+          description: >-
+            Advertise A-D Routes per EVI.
+          type: boolean
+          default: true
+
+    Bgp.EthernetSegment.DfElection:
```
Comment (manomugdha) [inline device/bgp/evpn/ethernetsegment.yaml:43]: why it is not under properties?

File: device/bgp/evpn/ethernetsegmentv4.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+components:
+  schemas:
+    Bgp.EthernetSegmentV4:
+      x-include:      
+      - '../../device.yaml#/components/schemas/Device.Active'
```
Comment (manomugdha) [inline device/bgp/evpn/ethernetsegmentv4.yaml:5]: should not it be part of base yaml?

File: device/bgp/evpn/ethernetsegmentv4.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      - './ethernetsegment.yaml#/components/schemas/Bgp.EthernetSegment'
+      description: >-
+        Configuration for BGP Ethernet Segment ranges.
+      type: object
+      properties:        
+        df_election:
+          description: >-
+            Designated Forwarder election configuration.
+          $ref: './ethernetsegment.yaml#/components/schemas/Bgp.EthernetSegment.DfElection'
```
Comment (manomugdha) [inline device/bgp/evpn/ethernetsegmentv4.yaml:14]: why it is not just a property of base yaml?

File: device/bgp/evpn/ethernetsegmentv6.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+components:
+  schemas:
+    Bgp.EthernetSegmentV6:
+      x-include:      
+      - '../../device.yaml#/components/schemas/Device.Active'
```
Comment (manomugdha) [inline device/bgp/evpn/ethernetsegmentv6.yaml:5]: same comments as v4.

File: device/bgp/evpn/eviroutedistinguisher.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+components:
+  schemas:
+    Bgp.EviRouteDistinguisher:
```
Comment (manomugdha) [inline device/bgp/evpn/eviroutedistinguisher.yaml:3]: should not we make it generic for bgp like RouteDistinguisher? and then whoever needs it just refer it. file should be renamed and placed outside evi (under bgp).

File: device/bgp/evpn/eviroutetarget.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+components:
+  schemas:
+    Bgp.EviRouteTarget:
```
Comment (manomugdha) [inline device/bgp/evpn/eviroutetarget.yaml:3]: same comment as RD

File: device/bgp/evpn/evivxlan.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
+components:
+  schemas:
+    Bgp.EviVxlan:
```
Comment (manomugdha) [inline device/bgp/evpn/evivxlan.yaml:3]: how evimpls and other will be incorporated?

File: device/bgp/evpn/evpnmacranges.yaml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    Bgp.CMACRanges:
+      description: >-
+        The MAC Ranges per BroadcastDomain
+      type: object
+      required: [mac_addresses]
+      properties:        
+        mac_addresses:
+          description: >-
+           Host MAC address range per Broadcast Domain 
+          $ref: '../../routes/routeaddresses.yaml#/components/schemas/MACRouteAddress'
```
Comment (manomugdha) [inline device/bgp/evpn/evpnmacranges.yaml:12]: should it be ref or include? today we support include.
may be good to support ref.

File: device/bgp/evpn/broadcastdomain.yaml [hunk 12]
```diff
@@ -0,0 +1,19 @@
+components:
+  schemas:
+    Bgp.BroadcastDomain:
+      description: >-
+        Broadcast Domain Per EVI Configuration.
+      type: object
+      properties:        
+        ethernet_tagid:
```
[the remark this answers is not in the document]
Comment (amrghosh) [reply device/bgp/evpn/broadcastdomain.yaml:8]: As per RF7432  its Ethernet tag ID

**Comment by amrghosh** [comment]
closing

**Pull request closed without merging**